pkg/problemdetector/problem_detector.go

```diff
@@ -47,17 +45,16 @@ func NewProblemDetector(monitors map[string]types.Monitor, exporters []types.Exp
 func (p *problemDetector) Run() error {
 	// Start the log monitors one by one.
 	var chans []<-chan *types.Status
-	for cfg, m := range p.monitors {
+	for _, m := range p.monitors {
 		ch, err := m.Start()
 		if err != nil {
 			// Do not return error and keep on trying the following config files.
-			glog.Errorf("Failed to start log monitor %q: %v", cfg, err)
+			glog.Errorf("Failed to start log monitor %v: %v", m, err)
 			continue
 		}
-		chans = append(chans, ch)
-	}
-	if len(chans) == 0 {
```

Comment:
Why is this removed?

Reply:
This logic is moved to "cmd/node_problem_detector.go" in this commit:
```
if len(problemDaemons) == 0 {
	glog.Fatalf("No problem daemon is successfully setup")
}
```

Reply:
Good point. Done.